### DISCUSSION TITLE
Open `file_path:line_number` file links from terminal (e.g. by clicking with your mouse on a file path displayed in terminal output)

### DISCUSSION THREAD

**Discussion by simon-liebehenschel** [NvChad/NvChad#2009] [Q&A] [answered]
## What

In almost any IDE (e.g. PyCharm, VSCode) you can click with your mouse on a file path in your terminal, and this file will be opened on exact line number in the IDE.

For example, you executed some command, and you see in your terminal the command output which mentions some files and line numbers:
```bash
$ pre-commit run --all-files
project/foobar.py:476:4: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
```
You click with your mouse on the `project/foobar.py:476` file link, and your IDE opens that file (`project/foobar.py`) on that line (line `476`).

## What I found by myself

I did a web search, but I still not sure whether this must be implemented on my Linux, or in nvim config, or via some plugin.

I've read NvChad documentation cover to cover, but I am not sure that any suitable plugin is included.

UPDATE: I found https://www.youtube.com/watch?v=6RxcXV9sl5s but this is still ugly to type manually `nvim +<LINE_NUMBER> <FILE_PATH>` for every link I need to open. I am still looking for a way to open links quickly.

## Question

Is there any plugin included in NvChad which can handle the described functionality?

**Comment by simon-liebehenschel** [comment, marked as answer]
This must be done by myself for my terminal. Finally I found something that looks like a solution https://github.com/artemave/tmux_super_fingers

**Comment by dinvlad** [comment]
I’m also interested in this. This feature seems very terminal-dependent, e.g. in it’s called Semantic History in iTerm2 and “hints” in Kitty. And in either of those at least, it only works locally, since it doesn’t have native integration with Neovim like it does in VS Code.

**Reply by simon-liebehenschel** [reply]
You can use Kitty terminal. Press [ctrl+shift+p>n](https://sw.kovidgoyal.net/kitty/conf/#shortcut-kitty.Open-the-selected-file-at-the-selected-line) and `file_path`:`line_number` will be opened in vim.
